code-challenges.test.js

```diff
@@ -10,37 +10,208 @@
+
+//  value.name[value.name.indexOf(" ") + 1]
+// this will access the first letter of the word after the first instance of an empty space
+
+// follow that with .toUpperCase() and then concat another slice, with the starting index being TWO indexes forward from the empty space index.
+
+// value.name[value.name.indexOf(" ") + 1].toUpperCase() +
+//     value.name.slice(value.name.indexOf(" ") + 2)
+
+// now that we have capitalized each word in the name key, all we have to do is use string interpolation, placing each super long method in ${} and then call on the occupation key ${value.occupation}. `${first} ${last} is {value.occupation}.`
+
+// make sure you put a return in front of your map, and your string interpolation.
+
+// then you can cry!!!
+
+// -------------------
+
+// this was the first pass, not very readable, and not dynamic for a name key with more than two words as a value.
+
+// const hitchhikersGuideToProgramming = (array) => {
+//   return array.map((value) => {
+//     return `${
+//       value.name[0].toUpperCase() + value.name.slice(1, value.name.indexOf(" "))
+//     } ${
+//       value.name[value.name.indexOf(" ") + 1].toUpperCase() +
+//       value.name.slice(value.name.indexOf(" ") + 2)
+//     } is ${value.occupation}.`;
+//   });
+//};
+
+// -------------------
+
+// second pass, assigned each method for each word to a varable; firstName and lastName, then used string interpolation with the variables, more readible, still not the most dynamic
+
+// const hitchhikersGuideToProgramming = (array) => {
+//   return array.map((value) => {
+//     const firstName =
+//       value.name[0].toUpperCase() +
+//       value.name.slice(1, value.name.indexOf(" "));
+//     const lastName =
+//       value.name[value.name.indexOf(" ") + 1].toUpperCase() +
+//       value.name.slice(value.name.indexOf(" ") + 2);
+//     return `${firstName} ${lastName} is ${value.occupation}.`;
+//   });
+// };
+
+// ---------------------
+
+// third pass, maps over array, and then splits the name keys into their own array, maps over that new array, and uses our earlier method without having to access the empty space index, cleaner and dynamic. I also tried out the .substring() method here to try something other than .slice(), still works.
+
+const hitchhikersGuideToProgramming = (array) => {
+  return array.map((value) => {
+    const capName = value.name
+      .split(" ")
+      .map((value) => {
+        return value[0].toUpperCase() + value.substring(1);
+      })
+      .join(" ");
+    return `${capName} is ${value.occupation}.`;
+  });
+};
+
+// ---------------------
+
+// final pass, regex, obviously i had to research the syntax here, and it doesn't really look that much more readable, unless you just accept that the regex is going to work as someone who didnt write this code. Basically it is just finding the first letter of each word in value.name and capitalizes only that letter.
```

Comment:
I agree, I think the 3rd function is more readable and a good approach. I like that you left all your other attempts in and it's really cool to see it evolve. Nicely done👏🏽